src/NServiceBus.Metrics.ServiceControl/ReportingFeature.cs

```diff
@@ -160,7 +160,7 @@ void SetUpOutgoingMessageMutator(FeatureConfigurationContext context, ReportingO
         {
             if (options.TryGetValidEndpointInstanceIdOverride(out var instanceId))
             {
-                context.Container.ConfigureComponent(() => new MetricsIdAttachingMutator(instanceId), DependencyLifecycle.SingleInstance);
+                context.Services.AddSingleton(new MetricsIdAttachingMutator(instanceId));
```

Comment:
should this be 

```suggestion
                context.Services.AddSingleton<IMutateOutgoingMessages>(new MetricsIdAttachingMutator(instanceId));
```

since it tries to register a mutator?

it would be better to just use the config.RegisterMutator API though given this doesn't really seem to need DI at all

Reply:
Good catch. This was picked up by a test.